### ISSUE TITLE
feat(evaluation): calculate validator-bundle marginal value with paired counterfactuals

### ISSUE THREAD

**Issue by everettVT** [VangelisTech/archetype#631]
## User story

As an operator—or an outer orchestration agent—improving the coding factory, I want critic findings and proposed validators evaluated against immutable historical candidates and held-out tasks with explicit confidence and cost measurements, so I can adopt only validators that add unique quality coverage or enable cheaper/faster author models without overfitting the task that suggested them.

Listing desirable metrics is not sufficient. This issue owns calculating them reproducibly.

## Product question

Given an existing validator bundle `B`, a proposed validator `v`, an author-model configuration `M`, and a fixed independent adjudication policy:

- Which bad candidates does `v` catch that `B` misses?
- Which acceptable candidates does `v` newly reject?
- How much execution and repair cost does it add?
- Is its coverage duplicated by existing validators?
- Does it generalize across held-out tasks, models, prompts, and samples?
- Can `B + v` let a smaller/faster author model meet the same repository-quality constraints?
- Which subset of proposals lies on the quality/cost Pareto frontier?

The optimization target is not “more validators” and not a lower quality threshold. It is:

```text
minimize expected dollars + weighted wall time

subject to:
  false-promotion risk upper bound ≤ epsilon
  state and behavior correctness ≥ required floor
  architecture compliance ≥ required floor
  maintainability and simplicity ≥ required floor
```

The initial subject is a repository candidate. “Artifact quality” is too broad for the first contract.

## Separation from the native critic

#630 owns candidate review and present-candidate repair.

This issue owns the outer learning loop:

```text
ReviewFinding
→ ValidatorProposal
→ shadow counterfactual experiment
→ held-out active experiment when warranted
→ explicit policy decision
→ versioned future validator bundle
```

A valid finding may block the current candidate without ever becoming a permanent validator. The critic or perspective that proposed a validator cannot install it, adjudicate its adoption alone, or edit the bundle used by its own experiment.

## Statistical definitions

For one immutable candidate:

- `Y = 1` means an independent adjudication says the candidate satisfies the declared repository-quality vector.
- `B = 1` means the baseline validator bundle passes.
- `v = 1` means the proposed validator passes.

Calculate and retain both counts and rates:

| Quantity | Definition |
|---|---|
| Absolute false-promotion risk | `P(Y = 0 and B = 1)` |
| Conditional false-promotion rate | `P(Y = 0 given B = 1)` |
| Proposed-validator catch rate | `P(v = 0 given Y = 0)` |
| Marginal catch | `P(Y = 0, B = 1, v = 0)` |
| Incremental coverage | `P(v = 0 given Y = 0, B = 1)` |
| Marginal false rejection | `P(Y = 1, B = 1, v = 0)` |
| Mutation kill rate | applicable non-equivalent mutants killed / applicable non-equivalent mutants |
| Execution cost | added validator dollars, CPU/GPU time, and wall time per candidate |
| Repair burden | added author dispatches, model tokens, dollars, and elapsed time under active use |
| Stability | the same effects stratified by held-out task, author model, prompt/policy, and sample |

Do not report a rate without its numerator, denominator, and confidence interval. Conditional false-promotion rate may be undefined when no candidate promotes; absolute risk remains defined and must also be reported.

### 1. Paired shadow replay

Run baseline and proposed validators against the same immutable candidate snapshots without exposing proposed-validator output to the author.

The shadow corpus includes:

- accepted and rejected candidates;
- exact base/head/diff and validator-bundle identities;
- independently adjudicated quality dimensions;
- critic findings and reproduction evidence;
- known historical failures;
- applicable repository mutations/counterfactuals; and
- exact validator execution cost and environment identity.

Because every bundle sees the same candidate, compute paired differences rather than comparing unrelated aggregate runs.

Shadow replay estimates catch, false rejection, overlap, mutation kill, incremental coverage, and direct execution cost. It does not estimate author adaptation or repair burden.

### 2. Held-out active factorial experiment

Only promising proposals enter active use. Randomize held-out tasks and seeds across configurations such as:

```text
large author model + baseline bundle
small author model + baseline bundle
small author model + proposed bundle
small author model + selective critic escalation
```

Hold independent adjudication policy fixed across treatment arms. Record author turns, model calls, input/output tokens when available, validator/critic calls, retries, wall time, and monetary cost.

This stage estimates repair burden, time-to-acceptable-candidate, and whether stronger validation substitutes for author-model capability.

## Ground truth and multiple perspectives

Validator experiments require an adjudication surface independent of the bundle under test.

- Mechanical state/behavior oracles remain preferred where available.
- Multiple pinned critic/judge perspectives may contribute typed evidence.
- One reproduced critical defect cannot be outvoted by several shallow approvals.
- Original perspective receipts remain immutable; a synthesizer may cluster duplicate findings but cannot rewrite them.
- Disagreement, incomplete coverage, or insufficient evidence yields inconclusive rather than a guessed label.
- The proposal-generating critic is excluded from being the sole adjudicator of its proposal.

The quality target is a vector: behavioral correctness, state invariants, architecture, maintainability, and simplicity. Do not collapse it into one ambiguous “artifact quality” scalar.

## Confidence and sampling

Repeated candidates or repairs from one task are correlated. Confidence calculations must cluster or bootstrap by task, not treat every dispatch as independent.

At minimum:

- retain task-level strata and seeds;
- report paired confidence intervals for bundle differences;
- use one-sided upper confidence bounds for false-promotion constraints;
- mark estimates inconclusive when the denominator is too small;
- preserve held-out tasks that did not generate the proposal;
- keep originating-task results separate from generalization evidence; and
- record model/prompt/bundle interactions rather than averaging them away.

Zero observed false promotions does not prove zero risk. As an operator-facing calibration, approximately 300 independent promoted candidates with zero failures are needed for a one-sided 95% upper bound near 1% under the rule-of-three approximation. The implementation must report the exact supported interval method.

## Validator selection

Rank and select proposals by marginal value, not raw popularity:

- unique bad candidates caught beyond the baseline bundle;
- marginal false rejection;
- mutation/counterfactual coverage;
- recurring held-out failure classes;
- execution and active-repair cost;
- stability across models/prompts/samples; and
- correlation or duplication with other validators.

Initial selection may use a budgeted set-cover/Pareto procedure over the training corpus, but adoption requires held-out confirmation. A validator that duplicates an existing gate at positive cost has zero marginal coverage. A validator that only catches its originating mutation is overfit until held-out evidence says otherwise.

Policy outcomes are explicit and versioned:

- adopt into the default bundle;
- retain as conditional/advisory;
- keep in shadow mode for more evidence; or
- reject.

A `ValidatorPolicyDecision` is application-owned policy, not an `EvalReceipt` and not a critic conclusion.

## Durable experiment identity

Every trial/result must identify:

- proposal and originating finding(s);
- exact candidate, task, repository, base/head SHA, and diff digest;
- baseline and treatment validator-bundle versions/digests;
- validator code/config/environment identity;
- author model/driver/prompt/policy identity;
- critic/adjudicator policies and receipt digests;
- treatment assignment and seed;
- applicable mutation/counterfactual identity;
- raw binary outcomes and quality-vector conclusions;
- author, validator, and critic tokens/calls where available;
- direct and repair wall time and monetary cost where available;
- experiment corpus/training/holdout split identity; and
- estimator, interval method, implementation version, and resulting confidence bounds.

Missing cost or usage fields remain explicitly unknown; they are never silently treated as zero.

## Ownership

- `archetype.missions`
  - family-owned proposal/finding/candidate identities needed by the domain;
  - no statistical experiment orchestration or bundle adoption authority.
- `archetype.evaluation`
  - reusable experiment, treatment, metric, and estimator value contracts where generally applicable.
- `archetype.app.evaluation`
  - query-backed corpus construction, pinned grading, paired/statistical calculations, durable receipts, and experiment orchestration.
- explicit application policy/catalog boundary
  - versioned `ValidatorPolicyDecision` and future bundle activation.
- repository harness
  - consumes shipped APIs to run Archetype-on-Archetype counterfactuals and mutations;
  - production code never imports repository `evals/` or test tooling.
- #489
  - optional Pydantic Evals/Logfire adapter and coding-agent experiment projection;
  - hosted tooling is not required for local calculation.
- #482
  - may later expose read-only cross-run projections;
  - never owns experiment or promotion decisions.

## Current gaps

The existing `EvaluationService` can pin one world snapshot, run a versioned grader, and persist an `EvalReceipt`. It does not yet provide:

- normalized candidate/bundle treatment rows;
- paired cross-candidate estimators;
- task-clustered confidence intervals;
- shadow-validator execution;
- randomized active assignment;
- bundle-policy decisions; or
- complete coding-agent usage and monetary-cost evidence.

Extend the product evaluation surface only for demonstrated requirements above. Do not create a second mission authority or move repository harness code into `src/archetype/`.

## Contract card

- **Behavior:** calculate the unique quality lift, harm, cost, and confidence of validator proposals before changing future mission policy.
- **Owning layer:** evaluation values/application workflow plus an explicit policy decision boundary.
- **Normative source:** dataset/evaluation ontology, repository harness, Agent Missions critic contract from #630, and a focused validator-experiment section added by this work.
- **Existing executable oracle:** `EvaluationService` receipt identity, repository eval metrics, mission validator evidence, and mutation harness.
- **Invariants at risk:** independent adjudication, immutable treatment identity, no self-installing critic, held-out separation, cost honesty, and evaluation non-authority.
- **Required validation:** deterministic synthetic confusion matrices, paired/clustered estimator fixtures, overfit/duplicate/expensive validator counterfactuals, randomized assignment replay, and one real Archetype candidate corpus.
- **Documentation affected:** evaluation ontology/operator guide, repository harness, Agent Missions, and #489.

## Acceptance evidence

- [ ] One normalized experiment table contains immutable candidate, treatment, ground-truth, cost, and provenance fields.
- [ ] Deterministic fixtures reproduce exact counts/rates for false promotion, marginal catch, incremental coverage, and false rejection.
- [ ] Every rate reports numerator, denominator, estimator version, and confidence interval.
- [ ] Repeated repairs from one task are clustered rather than counted as independent tasks.
- [ ] A duplicate validator receives zero incremental coverage and loses to the cheaper baseline.
- [ ] An expensive validator with no unique catch is Pareto-dominated.
- [ ] A validator that kills only its originating mutation is not adopted without held-out evidence.
- [ ] Shadow replay never changes mission state or feeds the active author.
- [ ] Active factorial assignment is seeded, queryable, and replayable.
- [ ] Repair burden is calculated only from active treatment, not inferred from shadow replay.
- [ ] Smaller-model versus stronger-bundle treatments retain comparable independent adjudication and exact cost evidence.
- [ ] Missing usage/cost is represented as unknown, never zero.
- [ ] The proposal-generating critic cannot be the sole adoption adjudicator.
- [ ] `EvalReceipt` remains evidence; a separate versioned policy decision changes future validator bundles.
- [ ] Human-attention measurement is not required.
- [ ] One real held-out Archetype experiment publishes the candidate-level DataFrame and an aggregate Pareto report.
- [ ] Deterministic tests, focused repository capability evidence, and `make ci` pass.

## Non-goals

- Automatically adopting every critic suggestion.
- Treating popularity or majority vote as sufficient evidence.
- Requiring human-attention instrumentation in the first version.
- Defining mature task regimes before the data supports them.
- Making Logfire or Pydantic Evals the system of record.
- Allowing evaluation to advance mission state.
- Replacing deterministic repository validators with LLM judges.
- Running paid model sweeps in ordinary CI.
- Optimizing a single scalar called artifact quality.

## Related

- #630 — native exact-head critic and finding production.
- #489 — coding-agent evaluation adapter and durable receipt integration.
- #482 — future query-only analytics projections.
- #478 — repository verification-profile compilation.
- #629 — later multitask placement and convergence treatments.
